File: java/src/test/java/no/bouvet/workshop/solid/a1_output/CsvConsumer.java
package no.bouvet.workshop.solid.a1_output;

import java.io.File;
import java.util.LinkedList;
import java.util.List;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.notNullValue;
import static org.hamcrest.CoreMatchers.nullValue;
import static org.junit.Assert.assertThat;

public class CsvConsumer {

    private final List<String> lines;

    private File startCalled, endCalled;

    public CsvConsumer() {
        this.lines = new LinkedList<>();
    }

    public void visitStart(File file) {
        assertThat("visitStart was visited before", startCalled, nullValue(File.class));
        startCalled = file;
    }

    public void visitLine(String line) {
        lines.add(line);
    }

    public void visitEnd(File file) {
        assertThat("visitEnd was visited before", endCalled, nullValue(File.class));
        endCalled = file;
    }

    public void assertState() {
        assertThat("There were not 1000 lines visited as expected", lines.size(), is(1000));
        assertThat("Sample line one is not matching", lines.get(0), is("Country [code= \"AU\" , name=\"Australia\"]"));
        assertThat("Sample line 1000 is not matching", lines.get(999), is("Country [code= \"ES\" , name=\"Spain\"]"));
        assertThat("visitStart was not called", startCalled, notNullValue(File.class));
        assertThat("visitEnd was not called", endCalled, notNullValue(File.class));
    }
}

File: dotNet/delegation/test/a1_output/CsvConsumer.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.IO;

namespace test.a1_output
{
    public class CsvConsumer
    {
        private readonly List<string> lines;

        private FileInfo startCalled, endCalled;

        public CsvConsumer()
        {
            this.lines = new List<string>();
        }

        public void visitStart(FileInfo file)
        {
            Assert.IsNull(startCalled, "visitStart was visited before");
            startCalled = file;
        }

        public void visitLine(string line)
        {
            lines.Add(line);
        }

        public void visitEnd(FileInfo file)
        {
            Assert.IsNull(endCalled, "visitEnd was visited before");
            endCalled = file;
        }

        public void assertState()
        {
            Assert.AreEqual(1000, lines.Count, "There were not 1000 lines visited as expected");
            Assert.AreEqual("Country [code= \"AU\" , name=\"Australia\"]", lines[0], "Sample line one is not matching");
            Assert.AreEqual("Country [code= \"ES\" , name=\"Spain\"]", lines[999], "Sample line 1000 is not matching");
            Assert.IsNotNull(startCalled, "visitStart was not called");
            Assert.IsNotNull(endCalled, "visitEnd was not called");
        }
    }
}
